platform/view/services/endpoint/service.go

```diff
@@ -146,18 +147,14 @@ func (r *Service) GetResolver(ctx context.Context, id view.Identity) (*Resolver,
 	return r.resolver(ctx, id)
 }
 
-func (r *Service) Bind(ctx context.Context, longTerm view.Identity, ephemeral view.Identity) error {
-	if longTerm.Equal(ephemeral) {
-		logger.DebugfContext(ctx, "cannot bind [%s] to [%s], they are the same", longTerm, ephemeral)
-		return nil
+func (r *Service) Bind(ctx context.Context, longTerm view.Identity, ephemeralIDs ...view.Identity) error {
+	// filter out the ids that are equal to longTerm
+	toBind := collections.FilterSlice(ephemeralIDs, func(id view.Identity) bool {
+		return !longTerm.Equal(id)
+	})
```

Comment:
if `toBind` is empty, should we `return` here early with a proper error message?

Reply:
you are right, probably not, as we also just return nil if the ephemerals are empty.